src/main/kotlin/com/assisteddarwinism/ultimatesmashtracker/player/PlayerController.kt

```diff
@@ -20,32 +25,61 @@ class PlayerController {
     @Autowired
     lateinit var playerRepository: PlayerRepository
 
+    @Autowired
+    lateinit var tokenRepository: TokenRepository
+
+    @Autowired
+    lateinit var matchRepository: MatchRepository
+
+    @Autowired
+    lateinit var playerCharacterCombinationRepository: PlayerCharacterCombinationRepository
+
     @GetMapping
-    fun getPlayers(@RequestHeader("X-AuthToken") token: String) : List<Player> {
+    fun getPlayers(@RequestHeader("X-AuthToken") token: String): List<Player> {
         tokenValidator.checkTokenValid(token)
 
         return playerRepository.findAll().map { it -> Player(it.id!!, it.name) }
     }
 
     @GetMapping("/{playerId}")
-    @ResponseStatus(HttpStatus.NOT_IMPLEMENTED)
-    fun getPlayer(@RequestHeader("X-AuthToken") token: String, @PathVariable("playerId") playerId: Long){
+    fun getPlayerDetails(@RequestHeader("X-AuthToken") token: String, @PathVariable("playerId") playerId: Long): Player {
         tokenValidator.checkTokenValid(token)
+        if (!playerRepository.existsById(playerId)) throw ResourceNotFoundException()
+        val player = playerRepository.findById(playerId).get()
+        val matches = matchRepository.findAll()
+                .map { Match(it, playerCharacterCombinationRepository.findAllByMatchId(it.id!!)
+                        .map { PlayerCharacterRelation(it) }) }
+

```

Comment:
I feel like there is a more kotlin way of doing these iterable tasks without having to check the match emptiness first, but I'm not sure what it is.